utils/sysfs/sysfs_test.go

```diff
@@ -134,10 +134,178 @@ func TestIsCPUOnline(t *testing.T) {
 }
 
 func TestIsCPUOnlineNoFileAndCPU0MustBeOnline(t *testing.T) {
+	// Test on x86.
+	origIsX86 := isX86
+	defer func() {
+		isX86 = origIsX86
+	}()
+	isX86 = true
+
 	sysFs := NewRealSysFs()
-	online := sysFs.IsCPUOnline("./testdata/missing_online/node0/cpu0")
+	online := sysFs.IsCPUOnline("./testdata/missing_online/node0/cpu33")
+	assert.False(t, online)
+}
+
+func TestIsCPU0OnlineOnx86(t *testing.T) {
+	// Test on x86.
+	origIsX86 := isX86
+	defer func() {
+		isX86 = origIsX86
+	}()
+	isX86 = true
+
+	sysFs := NewRealSysFs()
+	online := sysFs.IsCPUOnline("path/is/irrelevant/it/is/zero/that/matters/cpu0")
 	assert.True(t, online)
+}
 
-	online = sysFs.IsCPUOnline("./testdata/missing_online/node0/cpu33")
+func TestCPU0OfflineOnNotx86(t *testing.T) {
+	// Test on arch other than x86.
+	origIsX86 := isX86
+	defer func() {
+		isX86 = origIsX86
+	}()
+	isX86 = false
+
+	sysFs := NewRealSysFs()
+	online := sysFs.IsCPUOnline("./testdata/missing_online/cpu0")
 	assert.False(t, online)
 }
+
+func TestIsCpuOnlineRaspberryPi4(t *testing.T) {
+	// Test on arch other than x86.
+	origIsX86 := isX86
+	defer func() {
+		isX86 = origIsX86
+	}()
+	isX86 = false
+	hotplugStates = "./testdata_rpi4/hotplug/states"
+
+	sysFS := NewRealSysFs()
+	online := sysFS.IsCPUOnline("./testdata_rpi4/cpu0")
+	assert.True(t, online)
+
+	online = sysFS.IsCPUOnline("./testdata_rpi4/cpu1")
+	assert.True(t, online)
+
+	online = sysFS.IsCPUOnline("./testdata_rpi4/cpu2")
+	assert.True(t, online)
+
+	online = sysFS.IsCPUOnline("./testdata_rpi4/cpu3")
+	assert.True(t, online)
+}
+
+func TestIsCpuOnlineGraviton2(t *testing.T) {
+	// Test on arch other than x86.
+	origIsX86 := isX86
+	defer func() {
+		isX86 = origIsX86
+	}()
+	isX86 = false
+
+	sysFS := NewRealSysFs()
+	online := sysFS.IsCPUOnline("./testdata_graviton2/cpu0")
+	assert.True(t, online)
+
+	online = sysFS.IsCPUOnline("./testdata_graviton2/cpu1")
+	assert.True(t, online)
+
+	online = sysFS.IsCPUOnline("./testdata_graviton2/cpu2")
+	assert.True(t, online)
+
+	online = sysFS.IsCPUOnline("./testdata_graviton2/cpu3")
+	assert.True(t, online)
+}
+
+func TestGetUniqueCPUPropertyCountOnRaspberryPi4(t *testing.T) {
+	// Test on arch other than x86.
+	origIsX86 := isX86
+	defer func() {
+		isX86 = origIsX86
+	}()
+	isX86 = false
+	hotplugStates = "./testdata_rpi4/hotplug/states"
```

Comment:
does `hotplugStates` also need to be reset like isX86 since it's a package var and being overriden here?